### PR TITLE
Hud, Fire, OnscreenBlaBla

### PR DESCRIPTION
**Pull request by erorcun**

### CODE REVIEW

**Review by Nick007J — APPROVED** [review]

File: src/core/Fire.cpp [hunk 1]
```diff
@@ -369,8 +388,39 @@ CFireManager::ExtinguishPoint(CVector point, float range)
...
+	}
+	if (fireI == NUM_FIRES)
+		return false;
+
+	CFire *fireToExtinguish = &m_aFires[fireI];
+	fireToExtinguish->m_fWaterExtinguishCountdown -= 0.012f * CTimer::GetTimeStep();
+	CVector steamPos = fireToExtinguish->m_vecPos +
+		CVector((CGeneral::GetRandomNumber() - 128) * 0.0155f,
+			(CGeneral::GetRandomNumber() - 128) * 0.0155f,
+			CGeneral::GetRandomNumber() * 0.005f);
```
Comment (ShFil119) [inline src/core/Fire.cpp:411]: maybe it's incaccurate pi (3.1) divided by 200?

File: src/render/Hud.cpp [hunk 2]
```diff
@@ -334,162 +426,185 @@ void CHud::Draw()
...
 
-				if ((TotalAmmo - AmmoInClip) > 9999)
-					Ammo = 9999;
-				else
-					Ammo = TotalAmmo - AmmoInClip;
-			}
+					if ((TotalAmmo - AmmoInClip) / 10 <= 9999)
+						Ammo = (TotalAmmo - AmmoInClip) / 10;
+					else
+						Ammo = 9999;
```
Comment (ShFil119) [inline src/render/Hud.cpp:451]: min?

File: src/render/Hud.cpp [hunk 3]
```diff
@@ -334,162 +426,185 @@ void CHud::Draw()
...
+				} else {
+					Clip = AmmoInClip;
 
-			sprintf(sTemp, "%d-%d", Ammo, Clip);
-		}
+					if ((TotalAmmo - AmmoInClip) > 9999)
+						Ammo = 9999;
+					else
+						Ammo = TotalAmmo - AmmoInClip;
```
Comment (ShFil119) [inline src/render/Hud.cpp:458]: also min?

File: src/render/Hud.cpp [hunk 4]
```diff
@@ -1758,18 +1855,42 @@ void CHud::SetBigMessage(wchar *message, int16 style)
...
 	if (!CReplay::IsPlayingBack()) {
+		for (int i = 0; i < 256; i++) {
+			m_HelpMessage[i] = 0;
+		}
+		for (int i = 0; i < 256; i++) {
+			m_LastHelpMessage[i] = 0;
+		}
+		for (int i = 0; i < 256; i++) {
+			m_HelpMessageToPrint[i] = 0;
+		}
```
Comment (ShFil119) [inline src/render/Hud.cpp:1869]: ARRAYSIZE?

File: src/render/Hud.cpp [hunk 5]
```diff
@@ -1758,18 +1855,42 @@ void CHud::SetBigMessage(wchar *message, int16 style)
...
+
 		CMessages::WideStringCopy(m_HelpMessage, message, 256);
 		CMessages::InsertPlayerControlKeysInString(m_HelpMessage);
+		if (m_HelpMessageState == 0 || !CMessages::WideStringCompare(m_HelpMessage, m_HelpMessageToPrint, 256)) {
+			for (int i = 0; i < 256; i++) {
+				m_LastHelpMessage[i] = 0;
```
Comment (ShFil119) [inline src/render/Hud.cpp:1875]: +1

**Review by ShFil119 — APPROVED** [review]
looks ok

**Comment by erorcun** [comment]
comments done

**Pull request merged**